### PR TITLE
13749 electron webpack shutdown fix

### PR DESCRIPTION
**Pull request by joel-ciq**
fix|feat|chore|docs|refactor|style|test: #13749 [CARD]

[Kanban link](https://chartiq.kanbanize.com/ctrl_board/48/cards/13749/details/)

**Description of change** Callback needs to be passed through to electron launcher from gulp task to exit npm on 'Quit'.
* Changes
Callback added to shutdown npm properly.

**Description of testing**
Markdown will convert the 1s to the appropriate number.
1. Step 1. checkout and link the seed project to: https://github.com/ChartIQ/finsemble-electron-adapter/tree/13749-electron-webpack-shutdown-fix
1. Step 2. run 'npm run dev' from seed.
1. Step 3. From Finsemble window quit from toolbar.
1. Step 4. npm should exit properly in the terminal you ran 'npm run dev' from.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -86,4 +86,4 @@
         "webpack-dev-middleware": "^1.12.0",
         "webpack-glob-entries": "^1.0.1"
     }
-}
\ No newline at end of file
+}
```
Comment (NinjaManatee) [inline package.json]: Probably should remove the new line to minimize changes to unrelated files.

Comment (joel-ciq) [reply package.json]: updated.

File: gulpfile.js [hunk 2]
```diff
@@ -401,7 +401,8 @@
 		},
 		launchElectron: done => {
 			let config = {
-				manifest: taskMethods.startupConfig[env.NODE_ENV].serverConfig
+				manifest: taskMethods.startupConfig[env.NODE_ENV].serverConfig,
+				onElectronClose: (code)=> process.exit(code)
```
[the remark this answers is not in the document]
Comment (joel-ciq) [reply gulpfile.js]: No, I don't think so. I don't see watchClose being set anywhere and there's a null check before it's ever called. So essentially I think it's dead code and safe to remove in other places.

@NinjaManatee should I remove it?

**Comment by NinjaManatee** [comment]
If you are sure it is dead code, then yes.


Andy Gardner
*Director, Solutions Engineering*
p: 800.821.8147  m: 540.808.5302
w: chartiq.com  e: andy@chartiq.com

File: gulpfile.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (joel-ciq) [reply gulpfile.js]: okay @NinjaManatee that's updated. I'd say we could really use unit tests for the seed project. Should I open a ticket in the backlog for test coverage?

**Comment by carterbs** [comment]
@joel-ciq this branch has the wrong base. Please change to 3.11.0 and we can decide where it goes later.

**Comment by joel-ciq** [comment]
@carterbs should be all set now.

File: gulpfile.js [hunk 3]
```diff
@@ -402,7 +397,8 @@
 
 			let config = {
 				manifest: cfg.serverConfig,
-				chromiumFlags: JSON.stringify(cfg.chromiumFlags),
+				onElectronClose: (exitCode)=> process.exit(exitCode),
```
Comment (amaroks) [inline gulpfile.js]:
```suggestion
				onElectronClose: process.exit,
```
I think that could work as well
